### PR TITLE
Allow price to be outdated by 2 * quote-interval

### PR DESCRIPTION
**Pull request by thomaseizinger**
Putting the threshold directly at the interval refresh rate is prone
to being triggered erroneously when the quote just hasn't updated yet.

Allowing for twice the refresh interval should fix this.

To allow for easier monitoring of this problem, we also print a log
including the timestamp of the quote.

### CODE REVIEW

**Review by luckysori — APPROVED** [review]
Okay, I believe in this solution.

File: daemon/src/lib.rs [hunk 1]
```diff
@@ -421,17 +421,25 @@ where
...
+            .format(&time::format_description::well_known::Rfc3339)
+            .context("Failed to format timestamp")?;
+
+        tracing::debug!(%order_id, %current_price, %quote_timestamp, "Proposing settlement of contract");
+
+        if latest_quote.is_older_than(QUOTE_INTERVAL_MINUTES.minutes() * 2) {
             anyhow::bail!(
                 "Latest quote is older than {} minutes. Refusing to settle with old price.",
-                QUOTE_INTERVAL_MINUTES
+                QUOTE_INTERVAL_MINUTES * 2
```
Comment (klochowicz) [inline daemon/src/lib.rs:435]: nit: this sort of duplication is prone to rot (if we ever decide to change the solution to change the multiplier, there's high likelyhood that the log won't get updated and will be actively misleading.

I'd suggest extracting `let quote_threshold = QUOTE_INTERVAL_MINUTES * 2;` and then using it in both places

**Review by klochowicz — APPROVED** [review]
this should please the affected users & the auto-taker too :) good one!

File: daemon/src/lib.rs (hunk 1 above)
Comment (thomaseizinger) [reply daemon/src/lib.rs:435]: Agreed. Will send a follow-up but merge this now so autotaker can close positions again!

**Pull request merged**